### PR TITLE
Leaf/b2.1 forgot password link on login page

### PR DESCRIPTION
**Pull request by owwix**
B2.1 Forgot Password Link on Login page #108

**Home page:** 
Screen Shot 2023-07-28 at 10 31 39 AM

**Home page after clicking Forgot Password button:**
Screen Shot 2023-07-28 at 10 32 08 AM

### CODE REVIEW

File: meshapp/src/home/login.tsx [hunk 1]
```diff
@@ -52,6 +53,18 @@ const SignUp = () => {
...
+  const [showForgotPassword, setShowForgotPassword] = useState(false);
+
+  const updateShowForgotPasswordState = () => {
+    if (showForgotPassword == false) {
+      setShowForgotPassword(true)
+    }
+    if (showForgotPassword == true) {
+      setShowForgotPassword(false)
+    }
+
```
Comment (rezatnoMsirhC) [inline meshapp/src/home/login.tsx:66]: 1. This should be done without if-statements. 
2. If you really want to keep the if-statements, this should be if/else. `showForgotPassword` cannot be both true and false. An interviewer probably wouldn't be happy if they saw this. 

https://softwareengineering.stackexchange.com/questions/12807/make-a-big-deal-out-of-true

Comment (owwix) [reply meshapp/src/home/login.tsx]: Just addressed this, thank you for that!

File: meshapp/src/home/login.tsx [hunk 2]
```diff
@@ -52,6 +53,13 @@ const SignUp = () => {
 };
 
 const Login = () => {
+
+  const [showForgotPassword, setShowForgotPassword] = useState(false);
+
+  const updateShowForgotPasswordState = () => {
+    setShowForgotPassword(prevState => !prevState)
```
Comment (rezatnoMsirhC) [inline meshapp/src/home/login.tsx:60]: Wasn't this based on `showForgotPassword` previously? Should it be based on `prevState`?

Comment (rezatnoMsirhC) [reply meshapp/src/home/login.tsx:60]: Oh nevermind I think I'm looking at the wrong section 😅

File: meshapp/src/home/logged-out/home.tsx [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
+  const [showForgotPassword, setShowForgotPassword] = useState(false);
+
+  const updateShowForgotPasswordState = () => {
+    if (showForgotPassword == false) {
+      setShowForgotPassword(true)
+    }
+    if (showForgotPassword == true) {
+      setShowForgotPassword(false)
+    }
+  }
```
Comment (rezatnoMsirhC) [inline meshapp/src/home/logged-out/home.tsx:25]: Looks like I forgot to comment the same thing here. Why was this code duplicated though? Can't the logic be moved out into a common function both files can use?

**Comment by rezatnoMsirhC** [comment]
Please provide some testing info/screenshots

File: meshapp/src/home/login.tsx (hunk 2 above)
Comment (owwix) [reply meshapp/src/home/login.tsx:60]: https://stackoverflow.com/questions/73993973/understanding-prevstate-in-react

File: meshapp/src/home/logged-out/home.tsx (hunk 3 above)
Comment (owwix) [reply meshapp/src/home/logged-out/home.tsx]: https://github.com/LetsMesh/Site/issues/269 created an issue to track this

**Review by rezatnoMsirhC — APPROVED** [review]

File: meshapp/src/home/logged-out/home.tsx [hunk 4]
```diff
@@ -59,7 +67,7 @@ export default function LoggedOutPage() {
               filter: "drop-shadow(4px 4px 3px rgba(0,0,0,.6))",
             }}
           >
-            {LoginInput()}
+            {showForgotPassword ? <ForgotPassword updateShowForgotPasswordState={updateShowForgotPasswordState} /> : <LoginInput updateShowForgotPasswordState={updateShowForgotPasswordState} />}
```
Comment (misslame) [inline meshapp/src/home/logged-out/home.tsx:70]: This seems repetitive.  Is there anyway to incorporate it into login?

**Review by misslame — CHANGES_REQUESTED** [review]
Is it possible to have forgot password logic & rendering be a part of the "login" component rather than the logic scattered about on the login screen & home screen?

File: meshapp/src/home/logged-out/home.tsx (hunk 4 above)
Comment (owwix) [reply meshapp/src/home/logged-out/home.tsx]: Refactored :)

**Review by misslame — APPROVED** [review]
:D looks good

**Pull request merged**